controllers/dogu_controller_int_test.go

```diff
@@ -349,36 +377,68 @@ var _ = Describe("Dogu Upgrade Tests", func() {
 		}()).To(BeTrue())
 
 		By("Update dogu crd with new version")
+		Expect(func() bool {
+			return getObjectFromCluster(ctx, installedLdapDoguCr, ldapFromDoguLookupKey)
+		}()).To(BeTrue())
 		Expect(func() bool {
 			installedLdapDoguCr.Spec.Version = ldapToVersion
 			err := k8sClient.Update(testCtx, installedLdapDoguCr)
 			return err == nil
 		}()).To(BeTrue())
 
+		setExecPodRunning(ctx, "ldap")
+
 		By("Check new image in deployment")
+		deploymentAfterUpgrading := new(appsv1.Deployment)
 		Eventually(func() bool {
-			deploymentAfterUpgrading := new(appsv1.Deployment)
 			ok := getObjectFromCluster(testCtx, deploymentAfterUpgrading, ldapFromDoguLookupKey)
 			return ok && strings.Contains(deploymentAfterUpgrading.Spec.Template.Spec.Containers[0].Image, ldapToVersion)
 		}, TimeoutInterval, PollingInterval).Should(BeTrue())
+		By("Check startup probe failure threshold in deployment")
+		Expect(deploymentAfterUpgrading.Spec.Template.Spec.Containers[0].StartupProbe.FailureThreshold).To(Equal(int32(60)))
 
-		deleteDoguCrd(ctx, installedLdapDoguCr, ldapFromDoguLookupKey, true)
+		deleteDoguCr(ctx, installedLdapDoguCr, ldapFromDoguLookupKey, true)
 
 		Expect(DoguRemoteRegistryMock.AssertExpectations(mockeryT)).To(BeTrue())
 		Expect(ImageRegistryMock.AssertExpectations(mockeryT)).To(BeTrue())
 		Expect(EtcdDoguRegistry.AssertExpectations(mockeryT)).To(BeTrue())
 	})
 })
 
-func installDoguCrd(ctx context.Context, doguCr *k8sv1.Dogu) {
+// setExecPodRunning can be necessary because the environment has no controllers to really start the pods,
+// therefore the dogu controller waits until timeout.
+func setExecPodRunning(ctx context.Context, doguName string) {
+	By("Simulate ExecPod is running")
+	podList := &corev1.PodList{}
+
+	Eventually(func() bool {
+		err := k8sClient.List(ctx, podList)
+		if err != nil {
+			return false
+		}
+		for _, pod := range podList.Items {
+			if strings.Contains(pod.Name, doguName+"-execpod") && pod.Status.Phase != corev1.PodRunning {
+				pod.Status.Phase = corev1.PodRunning
+				err := k8sClient.Status().Update(ctx, &pod)
+				if err != nil {
+					return false
+				}
+				return true
+			}
+		}
+		return false
+	}, TimeoutInterval, PollingInterval).Should(BeTrue())
+}
+
+func installDoguCr(ctx context.Context, doguCr *k8sv1.Dogu) {
 	Expect(k8sClient.Create(ctx, doguCr)).Should(Succeed())
 }
 
-func updateDoguCrd(ctx context.Context, doguCr *k8sv1.Dogu) {
+func updateDoguCr(ctx context.Context, doguCr *k8sv1.Dogu) {
 	Expect(k8sClient.Update(ctx, doguCr)).Should(Succeed())
 }
 
-func deleteDoguCrd(ctx context.Context, doguCr *k8sv1.Dogu, doguLookupKey types.NamespacedName, deleteAdditional bool) {
+func deleteDoguCr(ctx context.Context, doguCr *k8sv1.Dogu, doguLookupKey types.NamespacedName, deleteAdditional bool) {
```

Comment:
Didn't we want to remove the `deleteObjectFromCuster` calls since they don't do anything anyway?
```suggestion
func deleteDoguCr(ctx context.Context, doguCr *k8sv1.Dogu, doguLookupKey types.NamespacedName) {
```

Reply:
Lets keep them. We are planning to use a real cluster for integration tests. In this case we don't have to rewrite the functionality. We should comment this.